vaccine_feed_ingest/runners/az/pinal_ph_vaccinelocations_gov/normalize.py

```diff
@@ -0,0 +1,166 @@
+
+
+# address is loosely structured and inconsistent, so we're going to bash our
+# way through it, mostly parsing from the end of the string
+def _get_address(site: dict) -> Optional[schema.Address]:
+    if "address" not in site or not site["address"]:
+        return None
+
+    address = site["address"]
+    address = address.replace("        ", " ")
+    address = address.replace("    ", " ")
+    address = address.replace("    ", " ")
+    address = address.replace("  ", " ")
+    address = address.replace(" ,", ",")
+    address = address.replace(",,", ",")
+    address = address.lstrip()
+    address = address.rstrip()
+
+    # pull a zip code off the end
+    zip = None
+    if match := re.search(" (\\d\\d\\d\\d\\d-\\d\\d\\d\\d)$", address):
+        zip = match.group(1)
+        address = address.rstrip(f" {zip}")
+    if match := re.search(" (\\d\\d\\d\\d\\d)$", address):
+        zip = match.group(1)
+        address = address.rstrip(f" {zip}")
+
+    state = "AZ"
+    address = address.rstrip()
+    address = address.rstrip(",")
+    address = address.rstrip(".")
+    address = address.rstrip(f" {state}")
+    address = address.rstrip()
+    address = address.rstrip(",")
+    address = address.rstrip(f" {_get_city(site)}")
+    address = address.rstrip()
+    address = address.rstrip(",")
+
+    address_split = address.split(",")
+    street1 = address_split[0]
+    street2 = ", ".join(address_split[1:]) if len(address_split) > 1 else None
+
+    return schema.Address(
+        street1=street1,
+        street2=street2,
+        city=_get_city(site),
+        state=state,
+        zip=zip,
+    )
+
+
+def _get_contacts(site: dict) -> schema.Contact:
+    ret = []
+
+    if "phoneNumber" in site and site["phoneNumber"]:
+        raw_phone = str(site["phoneNumber"]).lstrip("1").lstrip("-")
+        if raw_phone[3] == "-" or raw_phone[7] == "-":
+            phone = "(" + raw_phone[0:3] + ") " + raw_phone[4:7] + "-" + raw_phone[8:12]
+        elif len(raw_phone) == 10:
+            phone = "(" + raw_phone[0:3] + ") " + raw_phone[3:6] + "-" + raw_phone[6:10]
+        else:
+            phone = raw_phone[0:14]
+
+        ret.append(schema.Contact(phone=phone))
+
+    if "website" in site and site["website"]:
+        ret.append(schema.Contact(website=site["website"]))
+
+    return ret
+
+
+def _get_inventories(site: dict) -> List[schema.Vaccine]:
+    ret = []
+    if "vaccineType" in site and site["vaccineType"]:
+        if "Moderna" in site["vaccineType"]:
+            ret.append(schema.Vaccine(vaccine=schema.VaccineType.MODERNA))
+        if "Pfizer" in site["vaccineType"]:
+            ret.append(schema.Vaccine(vaccine=schema.VaccineType.PFIZER_BIONTECH))
+        if "Janssen" in site["vaccineType"]:
+            ret.append(
+                schema.Vaccine(vaccine=schema.VaccineType.JOHNSON_JOHNSON_JANSSEN)
+            )
+    return ret
+
+
+def _get_organization(site: dict) -> Optional[schema.Organization]:
```

Comment:
Would this be a good place to use `provider_id_from_name` (from `vaccine_feed_ingest/utils/normalize.py`)?

Reply:
provider_id_from_name() had some very particular ideas about how names looked; I generalized to support the patterns in this set, then switched to using it. (This is the expected workflow going forward, right?)

Reply:
I don't think so, since that function assumes that there's a store number to be extracted -- that isn't true here.

Reply:
Thank you for trying! Sorry for the confusion.